_pytest/fixtures.py

```diff
@@ -1068,6 +1068,9 @@ def parsefactories(self, node_or_obj, nodeid=NOTSET, unittest=False):
         self._holderobjseen.add(holderobj)
         autousenames = []
         for name in dir(holderobj):
+            # Skip properties before the following getattr executes them.
+            if isinstance(type(holderobj).__dict__.get(name), property):
```

Comment:
What sort of cases are you thinking of? As I understand it, this line cannot throw, since:

- `type(holderobj)` always has a `__dict__` for all `holderobj`, and that `__dict__` is indeed a `dict` (as I mentioned in a comment above).

- Accessing the `__dict__` and checking `isinstance` can't trigger any Python voodoo.

On the other hand, using `safe_getattr` will still trigger voodoo, though it will continue if it throws.

That said, having properties on the class *type* is really getting farfetched, so using `safe_getattr` (or just `getattr`) is fine by me as well. Just let me know if you still want me to change it.